cluster_config_test.go

```diff
@@ -23,118 +24,178 @@ var ccfgTestJSON = []byte(`
 `)
 
 func TestLoadJSON(t *testing.T) {
-	cfg := &Config{}
-	err := cfg.LoadJSON(ccfgTestJSON)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if cfg.Peername != "testpeer" {
-		t.Error("expected peername 'testpeer'")
-	}
-
-	if cfg.ReplicationFactorMin != 5 {
-		t.Error("expected replication factor min == 5")
-	}
-
-	if !cfg.DisableRepinning {
-		t.Error("expected disable_repinning to be true")
-	}
-
-	j := &configJSON{}
-
-	json.Unmarshal(ccfgTestJSON, j)
-	j.ID = "abc"
-	tst, _ := json.Marshal(j)
-	err = cfg.LoadJSON(tst)
-	if err == nil {
-		t.Error("expected error decoding ID")
-	}
-
-	j = &configJSON{}
-	json.Unmarshal(ccfgTestJSON, j)
-	j.Peername = ""
-	tst, _ = json.Marshal(j)
-	err = cfg.LoadJSON(tst)
-	if cfg.Peername == "" {
-		t.Error("expected default peername")
-	}
-
-	j = &configJSON{}
-	json.Unmarshal(ccfgTestJSON, j)
-	j.PrivateKey = "abc"
-	tst, _ = json.Marshal(j)
-	err = cfg.LoadJSON(tst)
-	if err == nil {
-		t.Error("expected error parsing private key")
-	}
-
-	j = &configJSON{}
-	json.Unmarshal(ccfgTestJSON, j)
-	j.ListenMultiaddress = "abc"
-	tst, _ = json.Marshal(j)
-	err = cfg.LoadJSON(tst)
-	if err == nil {
-		t.Error("expected error parsing listen_multiaddress")
-	}
-
-	j = &configJSON{}
-	json.Unmarshal(ccfgTestJSON, j)
-	j.Secret = "abc"
-	tst, _ = json.Marshal(j)
-	err = cfg.LoadJSON(tst)
-	if err == nil {
-		t.Error("expected error decoding secret")
-	}
-
-	j = &configJSON{}
-	json.Unmarshal(ccfgTestJSON, j)
-	j.ReplicationFactor = 0
-	j.ReplicationFactorMin = 0
-	j.ReplicationFactorMax = 0
-	tst, _ = json.Marshal(j)
-	cfg.LoadJSON(tst)
-	if cfg.ReplicationFactorMin != -1 || cfg.ReplicationFactorMax != -1 {
-		t.Error("expected default replication factor")
-	}
-
-	j = &configJSON{}
-	json.Unmarshal(ccfgTestJSON, j)
-	j.ReplicationFactor = 3
-	tst, _ = json.Marshal(j)
-	cfg.LoadJSON(tst)
-	if cfg.ReplicationFactorMin != 3 || cfg.ReplicationFactorMax != 3 {
-		t.Error("expected replicationFactor Min/Max override")
-	}
-
-	j = &configJSON{}
-	json.Unmarshal(ccfgTestJSON, j)
-	j.ReplicationFactorMin = -1
-	tst, _ = json.Marshal(j)
-	err = cfg.LoadJSON(tst)
-	if err == nil {
-		t.Error("expected error when only one replication factor is -1")
-	}
-
-	j = &configJSON{}
-	json.Unmarshal(ccfgTestJSON, j)
-	j.ReplicationFactorMin = 5
-	j.ReplicationFactorMax = 4
-	tst, _ = json.Marshal(j)
-	err = cfg.LoadJSON(tst)
-	if err == nil {
-		t.Error("expected error when only rplMin > rplMax")
-	}
-
-	j = &configJSON{}
-	json.Unmarshal(ccfgTestJSON, j)
-	j.ReplicationFactorMin = 0
-	j.ReplicationFactorMax = 0
-	tst, _ = json.Marshal(j)
-	err = cfg.LoadJSON(tst)
-	if cfg.ReplicationFactorMin != -1 || cfg.ReplicationFactorMax != -1 {
-		t.Error("expected default replication factors")
-	}
+	loadJSON := func(t *testing.T) (*Config, error) {
```

Comment:
I have mixed feelings about this refactor. It's longer, more difficult to read and understand, more difficult to copy-paste, just looks nicer on the output but that's not really helpful anyway.

It's good to not copy paste though, but I feel the overall result is not improved.

Reply:
I find it easier to read because I can read it as just the contents of the `t.Run(...)` and not have to worry about whatever came before a particular test.

Reply:
well, you have to worry about jumping to see what `loadJSON2` actually does.